src/sqlfluff/core/dialects/dialect_ansi.py

```diff
@@ -1191,18 +1191,21 @@ class CaseExpressionSegment(BaseSegment):
                     "NOT",
                     "PRIOR",  # used in CONNECT BY clauses (EXASOL, Snowflake, Postgres...)
                 ),
-                Ref("Expression_A_Grammar"),
+                Ref("Expression_C_Grammar"),
```

Comment:
Hopefully this will reduce the incidence of those hard to reproduce recursion depth errors